api/src/main/java/io/strimzi/api/kafka/model/KafkaJmxOptionsAuthentication.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.strimzi.crdgenerator.annotations.Description;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+ * Configure the authentication protocol on the Kafka Broker's JMX port.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.EXISTING_PROPERTY,
+        property = "type")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+@JsonSubTypes({
+        @JsonSubTypes.Type(name = KafkaJmxOptionsAuthenticationPassword.TYPE_PASSWORD, value = KafkaJmxOptionsAuthenticationPassword.class)
+})
+public abstract class KafkaJmxOptionsAuthentication implements UnknownPropertyPreserving, Serializable {
```

Comment:
Tbh it should be just `KafkaJmxAuthentication` without the `Options` in the name.
I raised the point to call the class `KafkaJmxOptions` instead of `KafkaJmxRemote` because they are actually "options" for the JMX endpoint; in this case it's about "authentication for the JMX endpoint, not "authentication" for JMX options :-)